src/services/TechRecordsService.ts

```diff
@@ -136,6 +136,7 @@ class TechRecordsService {
     if (!this.validateVrms(techRecord)) {
       return Promise.reject({statusCode: 400, body: "Primary or secondaryVrms are not valid"});
     }
+    techRecord.techRecord[0] = isPayloadValid.value;
```

Comment:
Because the autopopulation is done using JOI. If you look in HgvValidation.ts there are two functions there, populateVehicleClassCode and populateBodyTypeCode which will autopopulate those fields. After checking if the payload is valid, I have to get the back the validated and autopopulated payload, and that is available on isPayloadValid.value

Reply:
Ok, makes sense. Because the initial implementation of populateVehicleClassCode and populateBodyTypeCode was done in 10209, it wasn't clear where those methods were getting used. All I could see was the raw dates which seemed weird.